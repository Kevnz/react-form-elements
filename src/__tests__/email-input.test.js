import React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'

import EmailInput from '../email-input'

describe('EmailInput component', () => {
  afterEach(() => {
    cleanup()
  })
  describe('The Rendered HTML', () => {
    it('should have a label', () => {
      const { getByText, container } = render(
        <EmailInput label="My EmailInput Label" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const label = getByText('My EmailInput Label')
      expect(label.textContent).toBe('My EmailInput Label')
    })
  })
  describe('handling input', () => {
    it('should set a value when passed in but not be valid', () => {
      const { getByLabelText, container } = render(
        <EmailInput label="My Example Email" initialValue="name@example.com" />
      )

      expect(container.firstChild).toMatchSnapshot()
      const input = getByLabelText('My Example Email')
      expect(input.value).toBe('name@example.com')
      expect(input.validity.valid).toBe(true)
      fireEvent.change(input, { target: { value: 'Updated Value' } })
      expect(input.validity.valid).toBe(false)
    })
  })
})
