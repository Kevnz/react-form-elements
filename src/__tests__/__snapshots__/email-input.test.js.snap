// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailInput component The Rendered HTML should have a label 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-1-1"
  >
    My EmailInput Label
  </label>
  <input
    id="id-1-1"
    name="TextBox"
    type="email"
    value=""
  />
</div>
`;

exports[`EmailInput component handling input should set a value when passed in but not be valid 1`] = `
<div
  class="rfe-form__row"
>
  <label
    for="id-2-1"
  >
    My Example Email
  </label>
  <input
    id="id-2-1"
    name="TextBox"
    type="email"
    value="name@example.com"
  />
</div>
`;
